src/compiler/scala/tools/nsc/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -1361,21 +1360,27 @@ abstract class BCodeBodyBuilder extends BCodeSkelBuilder {
     def genLoadTry(tree: Try): BType
 
     def genInvokeDynamicLambda(canLMF: delambdafy.LambdaMetaFactoryCapable) = {
-      import canLMF._
+      import canLMF.{ lambdaTarget => originalTarget, _ }
 
-      val isStaticMethod = lambdaTarget.hasFlag(Flags.STATIC)
+      val lambdaTarget = originalTarget.attachments.get[JustMethodReference].map(_.lambdaTarget).getOrElse(originalTarget)
       def asmType(sym: Symbol) = classBTypeFromSymbol(sym).toASMType
 
       val isInterface = lambdaTarget.owner.isTrait
```

Comment:
@dwijnand This needs to be generalized to `isTraitOrInterface` now that lambda implementation methods may be owned by Java interfaces.

Reply:
https://github.com/scala/scala/pull/9178